### ISSUE TITLE
Feature idea: Same cards, multiple column sets

### ISSUE THREAD

**Issue by kentonv** [libreboard/libreboard#211]
I've been using Libreboard to organize the roadmap for Sandstorm.io, and I thought of what I think would be an incredibly powerful feature.

The basic problem right now is that there are several different ways in which I might want to organize my cards. For example:

- Completion state (the usual wishes / not started / in progress / needs polish / done)
- Who is assigned to do the task
- Category
- How long it will take
- Milestones
- Priority

Right now, some of the above is handled by labels or by assignments, but the UI for these is far less pleasant than dragging cards around. Setting a label requires 5 precision clicks and delay time while UI elements load. Dragging cards between columns is way more fun.

So what I want instead is to be able to define multiple column sets for the same set of cards. Then, instead of setting labels or assignments, I would switch to the column set I am thinking about at the moment and push cards around.

Moreover, when I'm looking at one column set, the cards should show internal labels indicating the card's position is column sets I'm not currently viewing. So, e.g., when I'm looking at "completion state" I can also see each card's "priority" shown inside the card. (It may be nice to be able to choose exactly which labels are shown in order to avoid clutter... not sure. It would also be nice if I could color-code the labels by assigning colors to the columns from which they are derived.)

With this feature implemented, the usual label and assignments features could potentially be removed, simplifying the code and further differentiating from Trello.

What do you think?

**Comment by ocdtrekkie** [comment]
I, for one, think this sounds really cool.

**Comment by snolahc** [comment]
Hi,

This is an excellent idea, to abstract boards and cards, allowing multiple organizing schemas over the same cards.

Libreboard is the best tool to create lists.
In order to organize lists, we need to make lists of lists with priorities.
This tool allows us to be totally free on our organization, but could suggest a way to easily have a "todo list" from several boards where cards are assigned to me.

Managing those "duplicated/meta cards" would affect the underlying boards, cleaning them of the "done" stuff. This would be another board collecting cards with specific attributes/tags/...

PS: i created an issue but closed it after reading yours :)

**Comment by UltimaTom** [comment]
The idea is very interesting/original, but i'm not sure about the use case, plus it may complexify the UI and the usability. 

For now I use Libreboard this way : 
- Completion state :arrow_right: I use the columns
- Who is assigned to do the task :arrow_right: Name initials or avatar + filters
- Category :arrow_right: I use labels
- How long it will take :arrow_right: I don't use that :p 
- Milestones :arrow_right: I use the columns too ("Done for 0.1" "Done for 0.2" )
- Priority :arrow_right: I use the order of cards, top cards are higher priority

Furthermore, a task shouldn't be changing category, assignment, duration or priority that often, compared to the completion state IMO.

**Comment by kentonv** [comment]
@UltimaTom - Sure, not everyone needs this feature, obviously. But some people have different needs.

To address a few specific points:

- "Milestones" is supposed to mean "at what milestone does this need to be done", so it doesn't fit into the same columns as "completion state".
- "How long it will take" is in fact an important thing for me to visualize, because when there's month-long items next to single-day items, it can be misleading.
- Even if they don't change often, setting labels and assignments in the first place is quite tedious. When there's 200 cards that need to be categorized, it's easy to say "never mind, I don't want to do this", which is sad. Dragging between columns is much more pleasant.

**Comment by ocdtrekkie** [comment]
The other interesting thing, is @kentonv's suggestion produces all the necessary info, potentially, to output things like Gantt charts. You could manage things simply and easily as one expects with this sort of app, and view the data in any number of traditional project planning methodologies.

There's a lot of potential power to the concept of allowing the app to understand what it's doing (planning a project) rather than just being "lists of cards".

**Comment by xavierpriour** [comment]
I love the idea too, it is definitely something I'm missing in Trello right now!
To me, the best UI would be something like [HyperPlan](http://www.hyperplan.com/): you have attributes to the cards, and can choose which one to use as vertical/horizontal axis. Incredibly powerful.

**Comment by mikkelking** [comment]
I agree, I extended MypersonalKanban (written in Angular) to do precisely this, and the result is fantastic, because as kentonv says, it allows you to visualise the cards and manipulate them from different perspectives. One moment you are planning what will go in the next few releases of product, and the next moment you are balancing the workload between team members. I can put up a demo or some screen shots if anyone is interested to see it

**Comment by kentonv** [comment]
@mikkelking Would you be interested in making it a Sandstorm.io app? :) We (sandstorm team) are happy to help; come talk to us on [our discussion group](https://groups.google.com/group/sandstorm-dev) or IRC (#sandstorm on Freenode).

**Comment by kadishmal** [comment]
Have a look at how Asana can assign a single task to multiple projects and multiple sections. The idea is this: there are different people in the organization with different needs and views. Some people need to know the overview of the things going on in the company, some need to know the details. Say CTO doesn't need to know which tasks go into your next release, however he needs to see your task/card that tells the story of you next release. There are marketing guys who also don't care about implementation details, however they are interested in the parent/bigger tasks that their team and their next big launch depend on. So different people may want to choose to organize the same set of tasks/cards in different ways. The implementation that Asana team has provided is quite neat from the UX perspective and doesn't clutter the UI. Implementation wise it is relatively simple, too. Moreover, the simpler you think, the easier will be to move on and make additional changes in the future. Here is how Asana team has implemented this feature.

**Comment by maximd** [comment]
I'm commenting so I can subscribe to this issue alone, [since I can't do that by subscribing (i.e. watching) ~ #283](https://github.com/isaacs/github/issues/283)

**Comment by GustavoMarcante** [comment]
I think this feature is great. Gives you the possibility to use Kanban in many different ways.
I'd like to suggest to take a look at [ZenKit](https://zenkit.com/) and the way they made it happen.
I showed to my team, of "non tech people", and they liked it a lot. Very clear, easy and useful.
